### DISCUSSION TITLE
How to validate a field that is either string or array?

### DISCUSSION THREAD

**Discussion by galaczi** [adonisjs/core#3760] [Help] [answered]
I have a dynamic form where the user can submit one field or dynamically create more. How would I validate a field that is either a string or an array? Should I simply check the type of field and call a separate validator for each case?

**Comment by WeyIin** [comment, marked as answer]
What I would do is maybe iterate through the array and call the validator on that field with existing one..

**Reply by galaczi** [reply]
Hi Weylin, thanks for the idea.

What I have tried so far is expecting an array in the validator, and converting non-array values to an array before validating.

```
    const body = request.body()
    if(!Array.isArray(body.subject)) {
      body.subject = [body.subject]
      body.pricingMinutes = [body.pricingMinutes]
      body.pricingPrice = [body.pricingPrice]
      body.level = [body.level]
    }
```
Yet when I call ``request.validate``, the array validation still fails. Any idea why?

**Reply by WeyIin** [reply]
I think when you do `request.validate` it is not getting the `body` you are trying to validate.

Maybe using `request.updateBody(body)` at the end would do the trick?

**Reply by galaczi** [reply]
It worked! Thank you!
